pyproject.toml

```diff
@@ -0,0 +1,29 @@
+[build-system]
+requires = ["setuptools>=45", "setuptools_scm[toml]==7.*"]
+build-backend = "setuptools.build_meta"
+
+[project]
```

Comment:
```suggestion
[project]
requires-python = ">3.9"
```

Reply:
I see the CI accepts 3.7 but I think 3.9+ is more realistic.

Reply:
Does it not run on <3.9 ? otherwise no need to add artifical constraints (I think setting it like that completely prevents you from installing)

I think we can leave that out, anyway at some point we'll have to define what version of django/drf version we support, which will set the supported version of python anyway

Reply:
If we use an old Python version, the day we add a single type hint that is not compatible with that version, we have a runtime exception. For me the question is not so much "does it run" but "what expectations are we creating in the user in terms of our ability to support them". I really think creating the expectation that we can support 3.7 -> 3.11 is going to cause unnecessary trouble...

Reply:
For context, every major Linux distro is now defaulting to 3.10, and I think the Python toolchain on Windows is heavily leaning towards 3.9+.

Reply:
I'm not saying we should support 3.7 and 3.8 (this we can discuss later, at the same time when we clarify python requirements, test against a matrix of versions, etc). Just saying we should not artificially forbid pip-installing it (which I think is the effect of requires-python). We're developing a library and people may want to install it on existing django apps, running on existing setups, some of which not running the latest version of everything (and as you see in your matrix, 3.8 is still supported for some time !).

Reply:
Okay, so amended the commit suggestion to 3.8

Reply:
![image](https://user-images.githubusercontent.com/19554213/230358478-10104a15-c34f-4126-93a7-d307a10ebc16.png)